src/app/static/src/app/store/load/actions.ts

```diff
@@ -133,19 +133,34 @@ export const actions: ActionTree<LoadState, RootState> & LoadActions = {
 };
 
 const getRehydrateResult = async (context: ActionContext<LoadState, RootState>) => {
-    const {rootGetters, state, dispatch} = context
+    const {rootGetters, state, dispatch, rootState} = context
     const rehydrateId = state.rehydrateId
     const response = await api<LoadActionTypes, LoadErrorActionTypes>(context)
         .withSuccess("RehydrateResult")
         .withError("RehydrateResultError")
         .get<ProjectRehydrateResultResponse>(`rehydrate/result/${rehydrateId}`);
 
-    if (response && response.data && !rootGetters.isGuest) {
-        await (dispatch("projects/createProject",
-            {
-                name: state.projectName,
-                isUploaded: true
-            }, {root: true}));
+    if (response && response.data) {
+        const {files, savedState} = constructRehydrateProjectState(rootState, response.data)
+
+        if (!rootGetters.isGuest) {
+            await dispatch("projects/createProject",
+                {
+                    name: state.projectName,
+                    isUploaded: true
+                }, {root: true});
+
+            Object.assign(
```

Comment:
Again, I would prefer not to use Object.assign here, and I don't think using Object.assign for local objects to avoid null checks or other Typescript issues is an ideal approach. I think it's actually preferable to use `any` since at least makes it clear what we're doing. 

In this case, I guess the Typescript error is that this project state  is missing some properties? Could avoid that by creating our emptyState first then using emptyState.projects as the basis for the project state we assign to savedState?

Reply:
I don't see anything wrong in using Object.assign when updating for multiple properties of an object. I would go for equals assignment for its speed for single property assignment over object.assign.

You have also suggested Object.assign usage above with the spread operator `const newRootState = {...emptyState(), ...savedState}`, this is just a synthetic sagar for Object.assign.  However, I have updated the PR to effect your preference.

Reply:
It's more about the way it reads than the performance, and using it as a technique for avoiding Typescript errors, which I think is obfuscating.  Thanks for making the change.